### PR TITLE
Feat: Download icon is (only) visible (or enabled) when the batch is complete

### PR DESCRIPTION
**Pull request by johevemi**
## Description

Adds BatchHasBeenZipped which is a boolean field that tells if a batch has been zipped
There will be a PR in frontend domain as a follow up to this one

## References

* #17

### CODE REVIEW

File: source/dotnet/Contracts/BatchDtoV2.cs [hunk 1]
```diff
@@ -19,4 +19,4 @@ namespace Energinet.DataHub.Wholesale.Application.Batches;
 /// <summary>
 /// An immutable batch.
 /// </summary>
-public sealed record BatchDtoV2(Guid BatchNumber, DateTimeOffset PeriodStart, DateTimeOffset PeriodEnd, DateTimeOffset? ExecutionTimeStart, DateTimeOffset? ExecutionTimeEnd, BatchState ExecutionState);
+public sealed record BatchDtoV2(Guid BatchNumber, DateTimeOffset PeriodStart, DateTimeOffset PeriodEnd, DateTimeOffset? ExecutionTimeStart, DateTimeOffset? ExecutionTimeEnd, BatchState ExecutionState, bool BatchHasBeenZipped);
```
Comment (BjarkeMeier) [inline source/dotnet/Contracts/BatchDtoV2.cs:22]: How about?
```suggestion
public sealed record BatchDtoV2(Guid BatchNumber, DateTimeOffset PeriodStart, DateTimeOffset PeriodEnd, DateTimeOffset? ExecutionTimeStart, DateTimeOffset? ExecutionTimeEnd, BatchState ExecutionState, bool IsBasisDataDownloadAvailable);
```

File: source/dotnet/Services/Infrastructure/Persistence/Batches/BatchRepository.cs [hunk 2]
```diff
@@ -63,6 +63,13 @@ public async Task<List<Batch>> GetAsync(Instant minExecutionTimeStart, Instant m
             .ConfigureAwait(false);
     }
 
+    public async Task UpdateBatchHasBeenZippedToTrueAsync(Guid batchId)
+    {
+        var batch = _context.Batches.Single(b => b.Id == batchId);
+        batch.BatchHasBeenZipped = true;
+        await _context.SaveChangesAsync().ConfigureAwait(false);
+    }
```
Comment (BjarkeMeier) [inline source/dotnet/Services/Infrastructure/Persistence/Batches/BatchRepository.cs:71]: It's not part of the repository pattern to modify entities. This should be done outside. And also do not call `SaveChanges` but use `_unitOfWord.CommitAsync()` in your application service.

File: source/dotnet/Contracts/BatchDtoV2.cs (hunk 1 above)
Comment (BjarkeMeier) [reply source/dotnet/Contracts/BatchDtoV2.cs:22]: Suggestion applies throughout the PR.

Comment (johevemi) [reply source/dotnet/Contracts/BatchDtoV2.cs]: renamed to IsBasisDataDownloadAvailable

Comment (BjarkeMeier) [reply source/dotnet/Contracts/BatchDtoV2.cs]: nit: I'm sorry but `IsBasisDataZipAvailable` as the "download" thing is just one specific use case of the zip. You decide.

File: source/dotnet/Services/Infrastructure/Persistence/Batches/BatchRepository.cs (hunk 2 above)
Comment (johevemi) [reply source/dotnet/Services/Infrastructure/Persistence/Batches/BatchRepository.cs]: Done

File: source/dotnet/Services/DatabaseMigration/DatabaseMigration.csproj [hunk 3]
```diff
@@ -34,6 +34,8 @@ limitations under the License.
       <EmbeddedResource Include="Scripts\202209071400_Add_Batch_Period.sql" />
       <EmbeddedResource Include="Scripts\202209131440_Alter_Batch_Period.sql" />
       <EmbeddedResource Include="Scripts\202209191000_Add_Batch_Execution_Time.sql" />
+      <EmbeddedResource Include="Scripts\202210261230_Add_IsBatchDataDownloadAvailable.sql" />
+      <None Remove="Scripts\202210251400_Add_BatchHasBeenZipped.sql" />
```
Comment (BjarkeMeier) [inline source/dotnet/Services/DatabaseMigration/DatabaseMigration.csproj:38]: Remove this line?

**Review by BjarkeMeier — APPROVED** [review]

File: source/dotnet/Services/DatabaseMigration/DatabaseMigration.csproj (hunk 3 above)
Comment (johevemi) [reply source/dotnet/Services/DatabaseMigration/DatabaseMigration.csproj]: Ah yes, removed <None Remove="Scripts\202210251400_Add_BatchHasBeenZipped.sql" />

**Pull request merged**